target/linux/ar71xx/image/generic.mk

```diff
@@ -1,3 +1,20 @@
+define Device/OpenMesh
+  FILESYSTEMS := squashfs
+  DEVICE_PACKAGES := om-watchdog
+  KERNEL := kernel-bin | lzma | uImage lzma
+  IMAGES := sysupgrade.bin factory.bin
+  IMAGE/sysupgrade.bin/squashfs := append-rootfs | pad-rootfs | combined-ext-image
```

Comment:
Please use as last build step [append-metadata](https://git.lede-project.org/?p=source.git&a=search&h=HEAD&st=grep&s=append-metadata). You can drop the ```platform_check_image_target_openmesh()``` and most likely the  "Combined Extended Image v1" check as well in ```target/linux/ar71xx/base-files/lib/upgrade/openmesh.sh``` afterwards. The metadata (boardname appended to image) is validated against the boardname by ```package/base-files/files/lib/upgrade/fwtool.sh``` during sysupgrade.

If the boardname doesn't match for some reason, you can override the boardname that is used within the metadata using SUPPORTED_DEVICES. Have a look at d2b6bf141662429d542b23e9e8f44245a77710a5 for reference.

It might be worse to add an check for the the [maximum possible image size](https://git.lede-project.org/?p=source.git&a=search&h=HEAD&st=grep&s=check-size).